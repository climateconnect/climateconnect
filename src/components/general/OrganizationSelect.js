import React from "react";
import FormControl from "@material-ui/core/FormControl";
import InputLabel from "@material-ui/core/InputLabel";
import MenuItem from "@material-ui/core/MenuItem";
import Select from "@material-ui/core/Select";

export default function OrganizationSelect(props) {
  return (
    <FormControl fullWidth>
      <InputLabel id="organization-select-label">Organization</InputLabel>
      <Select
        labelId="organization-select-label"
        id="organization-select"
        value={props.value}
        onChange={props.handleChange}
        displayEmpty
        autoWidth
      >
        {organizationOptions(props.organizations)}
      </Select>
    </FormControl>
  );
}

function organizationOptions(organizations) {
  return organizations.map(organization => (
    <MenuItem key={organization.id} value={organization.id}>
      {organization.name}
    </MenuItem>
  ));
}
