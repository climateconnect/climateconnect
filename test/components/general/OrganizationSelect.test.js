import React from "react";
import OrganizationSelect from "../../../src/components/general/OrganizationSelect";
import renderer from "react-test-renderer";

describe("OrganizationSelect component", () => {
  let organizations = [
    {
      name: "sneep Erlangen e.V.",
      organization_image: "tmp-data/assets/images/sneep.png",
      id: 44
    },
    {
      name: "Climate Action Tech",
      organization_image: "tmp-data/assets/images/climate-action-tech.png",
      id: 45
    }
  ];
  it("renders each organization as a MenuItem", () => {
    const tree = renderer.create(
      <OrganizationSelect organizations={organizations}></OrganizationSelect>
    );
    expect(tree).toMatchSnapshot();
  });
});
