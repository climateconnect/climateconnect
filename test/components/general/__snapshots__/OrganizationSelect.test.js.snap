// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganizationSelect component renders each organization as a MenuItem 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
    data-shrink={false}
    id="organization-select-label"
  >
    Organization
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      aria-labelledby="organization-select-label organization-select"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
      id="organization-select"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
    </div>
    <input
      onAnimationStart={[Function]}
      type="hidden"
    />
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSelect-icon"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;
